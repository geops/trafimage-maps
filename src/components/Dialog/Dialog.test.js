import 'jest-canvas-mock';
import React from 'react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Map, View } from 'ol';
import ExportDialog from '.';

configure({ adapter: new Adapter() });

describe('ExportDialog', () => {
  const mockStore = configureStore([thunk]);
  let store;
  let map;

  beforeEach(() => {
    store = mockStore({
      map: {},
      app: {},
      oidc: { user: {} },
    });
    map = new Map({ view: new View({}) });
  });

  test('should match snapshot.', () => {
    const component = renderer.create(
      <Provider store={store}>
        <ExportDialog map={map} name="foo" />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
