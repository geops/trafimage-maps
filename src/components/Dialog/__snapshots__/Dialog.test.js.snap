// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportDialog should match snapshot. 1`] = `
<div
  className="wkp-dialog"
>
  <div
    className="tm-dialog"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="tm-dialog-container"
    >
      <div
        className="tm-dialog-header "
      >
        <div
          className="tm-header-left"
        />
        <div
          className="tm-button tm-dialog-close-bt"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
        >
          <svg
            fill="currentColor"
            focusable={false}
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
      </div>
      <div
        className="tm-dialog-body"
      />
    </div>
  </div>
</div>
`;
